docs/source/format/Columnar.rst

```diff
@@ -1385,6 +1385,36 @@ have two entries in each RecordBatch. For a RecordBatch of this schema with
     buffer 13: col2    data
 
 
+Compression
+-----------
+
+There are three different options for record batch body
+buffers compression: buffers can be uncompressed, can use
+``lz4`` or ``zstd`` compression codec. All buffers in the flat
+sequence of the message body are compressed separately with the
+same codec.
+
+The difference between compressed and uncompressed buffers in the
+serialized form is as follows:
+
+* If the buffers in the ``RecordBatch`` message are **compressed**
+
+  - the ``data header`` includes the length and memory offset
+    of each **compressed buffer** in the record batch's body
+
+  - the ``body`` includes a flat sequence of **compressed memory
+    buffers** together with the **length of uncompressed buffer**
+    stored in the first 8 bytes for each buffer in the sequence
```

Comment:
should we denote lsb or msb here?

Reply:
Something like "... **length of uncompressed buffer** as a 64-bit little-endian signed integer stored ..."?